### PR TITLE
Malware protection 6: Malware integration

### PR DESCRIPTION
**Pull request by mallexxx**
Task/Issue URL: https://app.asana.com/0/481882893211075/1208033567421351/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/1099

**Description**:
- add .malware threat and special page types

**Optional E2E tests**:
- [ ] Run PIR E2E tests
	Check this to run the Personal Information Removal end to end tests. If updating CCF, or any PIR related code, tick this.

**Steps to test this PR**:
0.1. Check out https://github.com/duckduckgo/content-scope-scripts/tree/pr-releases/pr-1268 and `npm install`; `npm run build`; add reference to the project
0.2. Check out https://github.com/duckduckgo/privacy-dashboard/tree/pr-releases/pr-250 and `npm install`; `npm run build`; add reference to the project 
 - or use the Review build

1. Validate http://privacy-test-pages.site/security/badware/malware.html detects malware
2. Validate http://privacy-test-pages.site/security/badware/phishing.html detects phishing
3. Validate Privacy Dashboard displaying status on overridden navigation
4. Validate Help/Report links (not ready yet)

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo/Tab/TabExtensions/SpecialErrorPageTabExtension.swift [hunk 1]
```diff
@@ -199,17 +204,16 @@ extension SpecialErrorPageTabExtension: SpecialErrorPageUserScriptDelegate {
 
     // Special error page "Leave site" button action
     func leaveSiteAction() {
-        guard webView?.canGoBack == true else {
-            webView?.close()
-            return
+        Task { @MainActor in
+            await self.webView?.openNewTabFromErrorPage()
```
Comment (SabrinaTardio) [inline DuckDuckGo/Tab/TabExtensions/SpecialErrorPageTabExtension.swift:208]: I don’t think SSL navigation was causing problems…
do we want to generalise this behaviour for all?

File: UnitTests/TabExtensionsTests/ErrorPageTabExtensionTest.swift [hunk 2]
```diff
@@ -195,7 +200,8 @@ final class ErrorPageTabExtensionTest: XCTestCase {
         errorPageExtention.leaveSiteAction()
 
         // THEN
-        XCTAssertTrue(mockWebView.goBackCalled)
+        XCTAssertTrue(closeTabCalled)
```
Comment (SabrinaTardio) [inline UnitTests/TabExtensionsTests/ErrorPageTabExtensionTest.swift:203]: These tests are Failing… also probably we should update the titles

**Review by SabrinaTardio — COMMENTED** [review]
I a couple of comments and questions!

File: DuckDuckGo/Localizable.xcstrings [hunk 3]
```diff
@@ -35845,6 +35845,18 @@
         }
       }
     },
+    "malware.error.page.tab.title" : {
```
[the remark this answers is not in the document]
Comment (mallexxx) [reply DuckDuckGo/Localizable.xcstrings:35848]: yes, as soon as we get our copy approved

File: DuckDuckGo/MaliciousSiteProtection/MaliciousSiteProtectionManager.swift [hunk 4]
```diff
@@ -46,11 +46,12 @@ extension MaliciousSiteProtectionManager {
     struct EmbeddedDataProvider: MaliciousSiteProtection.EmbeddedDataProviding {
 
         private enum Constants {
-            static let embeddedDataRevision = 1694418
-            static let phishingEmbeddedHashPrefixDataSHA = "d9eccd24d05ce16d4ab877574df728f69be6c7840aea00e1be11aeafffb0b1dc"
+            // TODO: Rollback the revision and filter (with -f) set when malware is available on the server
```
[the remark this answers is not in the document]
Comment (mallexxx) [reply DuckDuckGo/MaliciousSiteProtection/MaliciousSiteProtectionManager.swift:49]: yes, this is to be removed before merging

**Review by SabrinaTardio — APPROVED** [review]
Assuming that we want to have the same behaviour for SSL this looks good to me

**Pull request merged**